src/hyperfine/parameter_range.rs

```diff
@@ -83,16 +83,28 @@ fn build_parameterized_commands<'a, T: Numeric>(
     param_min: T,
     param_max: T,
     step: T,
+    command_names: Vec<&'a str>,
     command_strings: Vec<&'a str>,
     param_name: &'a str,
 ) -> Result<Vec<Command<'a>>, ParameterScanError> {
     validate_params(param_min, param_max, step)?;
     let param_range = RangeStep::new(param_min, param_max, step);
     let mut commands = vec![];
 
+    let mut i = 0;
+    let name_count = command_names.len();
     for value in param_range {
         for cmd in &command_strings {
+            // Sets the command name by index (remainder) if exists.
+            let name = if name_count > 0 {
+                Some(command_names[i % name_count])
```

Comment:
I think we should force the list of command names to have the correct size (and show an error otherwise). In that case, the modulo operation here shouldn't be necessary, right?

Reply:
Hi @sharkdp , sorry for the late reply. I tried to remember the problem when I implemented the code.

I think I tried to implement multiple command and command name Mapping for `--parameter-list` and `--parameter-scan` as below.

```
cargo run --
  'echo bar-{bar}' 'echo foo-{foo}'
  --parameter-list bar 1,2 --parameter-list foo 1,2
  --command-name Bar-{bar} --command-name Foo-{foo}
```

It could output the right result with this `remainder` code.

```
Benchmark #1: Bar-1

Benchmark #2: Foo-1

Benchmark #3: Bar-2

Benchmark #4: Foo-1

Benchmark #5: Bar-1

Benchmark #6: Foo-2

Benchmark #7: Bar-2

Benchmark #8: Foo-2
```

But this code really brings some issues if user specifies less `--command-name` than command size with `--parameter-list` and `--parameter-scan`.
Since the `--command-name` could be ignored or less than the command size. The missing command names are displayed as the shell command directly.

```
cargo run -- 'echo 1' 'echo 2' --command-name Name-1

Benchmark #1: Name-1

Benchmark #2: echo 2
```

But with this PR if user specifies less no-parameterized `--command-name` with `--parameter-scan`, it causes the issue as below. I think the third command name should be the shell command directly (not be `Name-1`).

```
cargo run -- 'echo {bar}' --parameter-scan bar 1 3 --parameter-step-size 1 --command-name Name-1 --command-name Name-2

Benchmark #1: Name-1

Benchmark #2: Name-2

Benchmark #3: Name-1
```

What do you think? Could you help give some suggestions about this problem? Thanks.

Reply:
Hmm. Ok. Let's try to make this a bit more formal.

In general, there can be:
- *C* commands
- *P* parameters with
- *V_m* values for each of the *P* parameters.

The total number of benchmarks B will be given by

*B = C * V_1 * V_2 * … * V_P*

Note that in your example (slightly modifiied) `'echo bar-{bar}' 'echo foo-{foo}' --parameter-list bar 1,2 --parameter-list foo 1,2,3` example, there would actually be 2 · 2 · 3 = 12 benchmarks, not just 2 · 3 = 6.

Up to now, users can specify `--command-name` up to *B* times. If the number of command names is less than *B*, the default generated name will be used. I think we should actually remove that freedom and implement the following behavior:

Either:
- `--command-name=<name>` appears exactly *B* times => use the i-th `<name>` for the i-th benchmark
Or:
- `--command-name=<name>` appears exactly once. In this case, *every* benchmark will use the `<name>`

In both cases, variable substitution will take place.

Reply:
@sorairolake Are you still interested in working on this? (no need for a rush. I'm only asking in case we need someone else to finish this).

Reply:
Sorry for late reply. I will continue working on this in this week.